vispy/app/backends/_tk.py

```diff
@@ -0,0 +1,701 @@
+def _set_config(c):
+    """Set gl configuration for template.
+    Currently not used for Tkinter backend.
+    """
+    return []
+
+
+# ------------------------------------------------------------- application ---
+
+_tk_inst = None         # Reference to tk.Tk instance
+_tk_inst_owned = False  # Whether we created the Tk instance or not
+_tk_toplevels = []      # References to created CanvasBackend Toplevels
+
+
+def _new_toplevel(self, *args, **kwargs):
+    """
+    Create and return a new withdrawn Toplevel.
+    Create a tk.Toplevel with the given args and kwargs,
+    minimize it and add it to the global list before returning.
+
+    :return: Return the created tk.Toplevel
+    """
+    global _tk_inst, _tk_toplevels
+    tl = tk.Toplevel(_tk_inst, *args, **kwargs)
+    tl.withdraw()
+    _tk_toplevels.append(self)
+    return tl
+
+
+def _del_toplevel(fr=None):
+    """
+    Destroy the given Toplevel, and if it was the last one,
+    also destroy the global Tk instance if we created it.
+
+    :param fr: The CanvasBackend to destroy, defaults to None
+    """
+    global _tk_inst, _tk_inst_owned, _tk_toplevels
+
+    if fr:
+        try:
+            fr.destroy()
+            if fr.top:
+                fr.top.destroy()
+            _tk_toplevels.remove(fr)
+        except Exception:
+            pass
+
+    # If there are no Toplevels left, quit the mainloop.
+    if _tk_inst and not _tk_toplevels and _tk_inst_owned:
+        _tk_inst.quit()
+        _tk_inst.destroy()
+        _tk_inst = None
+
+
+class ApplicationBackend(BaseApplicationBackend):
+    def __init__(self):
+        BaseApplicationBackend.__init__(self)
+
+    def _vispy_get_backend_name(self):
+        return tk.__name__
+
+    def _vispy_process_events(self):
+        """Process events related to the spawned Tk application window.
+        First, update the global tk.Tk() instance, then call `_delayed_update`
+        on every created Toplevel (to force a redraw), and process some Tkinter
+        GUI events by calling the Tk.mainloop and immediately exiting.
+        """
+        # Update idle tasks first (probably not required)
+        app = self._vispy_get_native_app()
+        app.update_idletasks()
+
+        # Update every active Canvas window
+        for c in _tk_toplevels:
+            c._delayed_update()
+
+        # Process some events in the main Tkinter event loop
+        # And quit so we can continue elsewhere (call blocks normally)
+        app.after(0, lambda: app.quit())
+        app.mainloop()
+
+    def _vispy_run(self):
+        """Start the Tk.mainloop. This will block until all Tk windows are destroyed."""
+        self._vispy_get_native_app().mainloop()
+
+    def _vispy_quit(self):
+        """Destroy each created Toplevel by calling _vispy_close on it.
+        If there are no Toplevels left, also destroy the global Tk instance.
+        """
+        global _tk_toplevels
```

Comment:
```suggestion
```

Reply:
Globals are moved to their own class `_TkInstanceManager`.